kusd/validator/states.go

```diff
@@ -69,6 +69,16 @@ func (val *Validator) notLoggedInState() stateFn {
 			}
 		}
 	} else {
+		// sanity check
+		isVoter, err := val.network.IsVoter(&bind.CallOpts{}, val.account.Address)
+		if err != nil {
+			log.Crit("Failed to verify the voter information", "err", err)
+			return nil
+		}
+		if !isVoter {
+			log.Crit("Invalid genesis - genesis validator needs to be registered as a voter", "address", val.account.Address)
```

Comment:
If it's not a voter, can still start validation at this point? I understand this branch of the code is for genesis validators that start after genesis validation (block > 0) and are not voters.

Reply:
All the validators that are not genesis, are not voters. They will enter the other condition and are forced to make a deposit. In the future we need to consider additional scenarios such as if a process crashes. (Related to the punishment issue)